catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/monitor/email/SubscribedWorkspaceNotifierTest.java

```diff
@@ -0,0 +1,163 @@
+    propertiesForMockQueryRequest.put("id", "1234abcd");
+    List<Serializable> valuesForMockAttribute = Collections.singletonList("test@email.com");
+    Set<String> metacardTagsForMockWorkspaceMetacard =
+        Collections.singleton(WorkspaceConstants.WORKSPACE_TAG);
+
+    when(mockFilterBuilder.allOf(any(Filter.class), any(Filter.class))).thenReturn(mock(And.class));
+    when(mockFilterBuilder.attribute(anyString())).thenReturn(mockAttributeBuilder);
+    when(mockAttributeBuilder.is()).thenReturn(mockExpressionBuilder);
+    when(mockExpressionBuilder.text(anyString())).thenReturn(mock(Filter.class));
+    when(mockQueryResponse.getResults()).thenReturn(resultsForMockQueryResponse);
+    when(mockQueryResponse.getRequest()).thenReturn(mockQueryRequest);
+    when(mockQueryRequest.getProperties()).thenReturn(propertiesForMockQueryRequest);
+    when(mockCatalogFramework.query(any(QueryRequest.class)))
+        .thenReturn(mockWorkspaceMetacardQueryResponse);
+    when(mockWorkspaceMetacardQueryResponse.getResults()).thenReturn(resultsForMockQueryResponse);
+    when(mockResult.getMetacard()).thenReturn(mockWorkspaceMetacard);
+    when(mockWorkspaceMetacard.getAttribute(Core.METACARD_OWNER)).thenReturn(mockAttribute);
+    when(mockWorkspaceMetacard.getMetacardType()).thenReturn(mock(MetacardType.class));
+    when(mockWorkspaceMetacard.getTags()).thenReturn(metacardTagsForMockWorkspaceMetacard);
+    when(mockAttribute.getValues()).thenReturn(valuesForMockAttribute);
+    when(mockSubjectIdentity.getUniqueIdentifier(any(Subject.class))).thenReturn("test@email.com");
+
+    subscribedWorkspaceNotifier =
+        new SubscribedWorkspaceNotifier(
+            mockSubjectIdentity, mockEmailNotifier, mockFilterBuilder, mockCatalogFramework);
+  }
+
+  @Test(expected = PluginExecutionException.class)
+  public void testProcessWithNullResponse() throws Exception {
+    subscribedWorkspaceNotifier.process(null);
+  }
+
+  @Test
+  public void testProcessWithNoResults() throws PluginExecutionException {
+    when(mockQueryResponse.getResults()).thenReturn(new ArrayList<Result>());
+    successfulProcess(false);
+  }
+
+  @Test
+  public void testProcessWithNoOwner() throws PluginExecutionException {
+    when(mockAttribute.getValues()).thenReturn(Collections.emptyList());
+    successfulProcess(false);
+  }
+
+  @Test
+  public void testProcessWhenOwnerIsNotUser() throws PluginExecutionException {
+    when(mockSubjectIdentity.getUniqueIdentifier(any(Subject.class)))
+        .thenReturn("testUser@email.com");
+    successfulProcess(false);
+  }
+
+  @Test
+  public void testProcessWithNullQueryId() throws PluginExecutionException {
+    when(mockQueryRequest.getProperties()).thenReturn(new HashMap<>());
+    successfulProcess(false);
+  }
+
+  @Test
+  public void testProcess() throws Exception {
+    successfulProcess(true);
+  }
+
+  private void successfulProcess(boolean sendEmail) throws PluginExecutionException {
```

Comment:
I would rename this function to `assertEmailSent` and the parameter to `expectedNumberOfSentEmails` and then use the variable in `times()` instead of this email paramter, it would make this more easy to understand.